### PR TITLE
docs: Update introspection

### PR DESCRIPTION
**Pull request by vpetrovykh**
Add casts and operators introspection sections.
Add TYPEOF and INTROSPECT operators.

### CODE REVIEW

File: docs/edgeql/funcops/type.rst [hunk 1]
```diff
@@ -144,3 +144,150 @@ Types
...
+
+
+-----------
+
+
+.. eql:operator:: TYPEOF: TYPEOF anytype -> type
+
+    :index: type introspect introspection
+
+    Static type inference operator
```
Comment (1st1) [inline docs/edgeql/funcops/type.rst]: Add a period.

File: docs/edgeql/funcops/type.rst [hunk 2]
```diff
@@ -144,3 +144,150 @@ Types
...
+
+
+-----------
+
+
+.. eql:operator:: INTROSPECT: INTROSPECT type -> Type
+
+    :index: type typeof introspection
+
+    Static type introspection operator
```
Comment (1st1) [inline docs/edgeql/funcops/type.rst]: Add period.

File: docs/edgeql/funcops/type.rst (hunk 1 above)
Comment (vpetrovykh) [reply docs/edgeql/funcops/type.rst]: OK

File: docs/edgeql/funcops/type.rst (hunk 2 above)
Comment (vpetrovykh) [reply docs/edgeql/funcops/type.rst]: OK

**Pull request merged**